### PR TITLE
Extend checks to Iris aux coords

### LINKED ISSUE #474: Non 1D longitudes are not forced to be 0-360
We are currently not forcing non 1D longitudes to be in 0-360 range. That is inconsistent and can lead to errors when working with irregular grids.

Pull request incoming to fix that

### PR DESCRIPTION
**Pull request by jvegasbsc**
Extend checks to coords that are added as auxiliary coordinates (i.e. latitude and longitude on irregular grids)

**Tasks**

-   [x] [Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do, if you haven't done so already (and add the link at the bottom)
-   [x] This pull request has a descriptive title that can be used in a changelog
-   [x] Add unit tests
-   [x] Public functions should have a numpy-style docstring so they appear properly in the [API documentation](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/api/esmvalcore.html). For all other functions a one line docstring is sufficient.
-   [x] If writing a new/modified preprocessor function, please update the [documentation](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/esmvalcore/preprocessor.html)
-   [x] Circle/CI tests pass. Status can be seen below your pull request. If the tests are failing, click the link to find out why.
-   [x] Codacy code quality checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why. If you suspect Codacy may be wrong, please ask by commenting.
-   [x] Please use `yamllint` to check that your YAML files do not contain mistakes
-   [x ] If you make backward incompatible changes to the recipe format, make a new pull request in the [ESMValTool repository](https://github.com/ESMValGroup/ESMValTool) and add the link below
* * *

Closes #474

### CODE REVIEW

File: esmvalcore/cmor/check.py [hunk 1]
```diff
@@ -431,8 +433,23 @@ def _check_coord_values(self, coord_info, coord, var_name):
...
+                new_bounds = coord.bounds.copy()
+                while new_lons.min() < 0:
+                    new_lons[new_lons < 0] += 360
+                    new_bounds[new_bounds < 0] += 360
+                while new_lons.max() > 360:
+                    new_lons[new_lons > 360] -= 360
+                    new_bounds[new_bounds > 360] -= 360
+                new_coord = coord.copy(new_lons, new_bounds)
+                dims = self._cube.coord_dims(coord)
+                self._cube.remove_coord(coord)
```
Comment (valeriupredoi) [inline esmvalcore/cmor/check.py]: this is not working as expected (see failed tests) - units need to change and you are adding eithet `lat` or `lon` as auxiliary coordinate without adding the other as auxiliary coordinate too

Comment (valeriupredoi) [reply esmvalcore/cmor/check.py]: actually, forget about it, the 2 failing tests are not well written for this new implementation: you are giving lon points to lat so of course it fails :grin:

**Comment by valeriupredoi** [comment]
OK fixed the tests, can you pls add a couple unit tests that should fail exactly on the things that are not failing on anymore after I fixed the tests? Ie points and bounds values outside the allowed range? :beer:

**Comment by jvegasbsc** [comment]
Required for ESMValGroup/ESMValTool#1079

Ready to merge from my side

**Comment by valeriupredoi** [comment]
As Murray Walker says: go go go

**Comment by mattiarighi** [comment]
I will try to have a look later today...

**Comment by mattiarighi** [comment]
Looks good to me, but I have limited experience with irregular grids.
Maybe @schlunma or @zklaus can have a look.

**Pull request merged**